SingularityExecutor/src/main/java/com/hubspot/singularity/executor/utils/DockerUtils.java

```diff
@@ -0,0 +1,125 @@
+package com.hubspot.singularity.executor.utils;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+import com.hubspot.singularity.executor.config.SingularityExecutorConfiguration;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.DockerException;
+import com.spotify.docker.client.messages.Container;
+import com.spotify.docker.client.messages.ContainerInfo;
+
+public class DockerUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(DockerUtils.class);
+
+
+  private final SingularityExecutorConfiguration configuration;
+  private final DockerClient dockerClient;
+  private final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+  @Inject
+  public DockerUtils(SingularityExecutorConfiguration configuration, DockerClient dockerClient) {
+    this.configuration = configuration;
+    this.dockerClient = dockerClient;
+  }
+
+  public int getPid(final String containerName) throws DockerException {
+    return inspectContainer(containerName).state().pid();
+  }
+
+  public boolean isContainerRunning(final String containerName) throws DockerException {
+    return inspectContainer(containerName).state().running();
+  }
+
+  public ContainerInfo inspectContainer(final String containerName) throws DockerException {
+    Callable<ContainerInfo> callable = new Callable<ContainerInfo>() {
+      @Override public ContainerInfo call() throws Exception {
+        return dockerClient.inspectContainer(containerName);
+      }
+    };
+
+    try {
+      return callWithTimeout(callable);
+    } catch (Exception e) {
+      LOG.error("Caught exception while getting container status", e);
```

Comment:
I would only log all of these if you don't control the code that is going to handle the thrown exception - meaning, you need to ensure it gets logged here because you don't know if it will be logged when caught.